snapcraft/tests/commands/test_collaborate.py

```diff
@@ -0,0 +1,105 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+from unittest import mock
+
+import fixtures
+
+from snapcraft.main import main
+from snapcraft import (
+    storeapi,
+    tests
+)
+
+
+class CollaborateTestCase(tests.TestCase):
+
+    command_name = 'collaborate'
+
+    def setUp(self):
+        super().setUp()
+        self.fake_logger = fixtures.FakeLogger(level=logging.INFO)
+        self.useFixture(self.fake_logger)
+        self.fake_store = tests.fixture_setup.FakeStore()
+        self.useFixture(self.fake_store)
+        self.client = storeapi.StoreClient()
+        self.fake_terminal = tests.fixture_setup.FakeTerminal()
+        self.useFixture(self.fake_terminal)
+        patcher = mock.patch('snapcraft._store.Popen')
+        self.popen_mock = patcher.start()
+        rv_mock = mock.Mock()
+        rv_mock.returncode = 0
+        rv_mock.communicate.return_value = [b'foo', b'']
+        self.popen_mock.return_value = rv_mock
+        self.addCleanup(patcher.stop)
+
+    def test_collaborate_success(self):
+        self.client.login('dummy', 'test correct password')
+        main([self.command_name, 'ubuntu-core'])
```

Comment:
Can do that but, 'ubuntu-core' is already a fake snap in other tests and I'm reusing them. Do you think it's best to have another fake snap with only name diff to do the same thing. (this is for happy path btw)